From c70ef244c862e8e13ff6552ee829e968186e0f45 Mon Sep 17 00:00:00 2001
From: Omair Majid <omajid@redhat.com>
Date: Fri, 8 Nov 2019 21:01:11 -0500
Subject: [PATCH] Add runtime id for Arch Linux

Like gentoo, Arch Linux is a rolling release and doesn't have explicit
version numbers.

    $ cat /etc/os-release
    NAME="Arch Linux"
    PRETTY_NAME="Arch Linux"
    ID=arch
    BUILD_ID=rolling
    ANSI_COLOR="0;36"
    HOME_URL="https://www.archlinux.org/"
    DOCUMENTATION_URL="https://wiki.archlinux.org/"
    SUPPORT_URL="https://bbs.archlinux.org/"
    BUG_REPORT_URL="https://bugs.archlinux.org/"
    LOGO=archlinux
---
 .../runtime.compatibility.json                  | 17 +++++++++++++++++
 .../Microsoft.NETCore.Platforms/runtime.json    | 11 +++++++++++
 .../runtimeGroups.props                         |  5 +++++
 3 files changed, 33 insertions(+)

diff --git a/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json b/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json
index 2a196ecc9fc7..d273634527ef 100644
--- a/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json
+++ b/pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json
@@ -270,6 +270,23 @@
     "any",
     "base"
   ],
+  "arch": [
+    "arch",
+    "linux",
+    "unix",
+    "any",
+    "base"
+  ],
+  "arch-x64": [
+    "arch-x64",
+    "arch",
+    "linux-x64",
+    "linux",
+    "unix-x64",
+    "unix",
+    "any",
+    "base"
+  ],
   "base": [
     "base"
   ],
diff --git a/pkg/Microsoft.NETCore.Platforms/runtime.json b/pkg/Microsoft.NETCore.Platforms/runtime.json
index e70f2da7e668..5b2f70f44b9e 100644
--- a/pkg/Microsoft.NETCore.Platforms/runtime.json
+++ b/pkg/Microsoft.NETCore.Platforms/runtime.json
@@ -121,6 +121,17 @@
         "any"
       ]
     },
+    "arch": {
+      "#import": [
+        "linux"
+      ]
+    },
+    "arch-x64": {
+      "#import": [
+        "arch",
+        "linux-x64"
+      ]
+    },
     "base": {
       "#import": []
     },
diff --git a/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props b/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
index 6c95b67819be..aa31cac61f34 100644
--- a/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
+++ b/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
@@ -25,6 +25,11 @@
       <Versions>21</Versions>
     </RuntimeGroup>

+    <RuntimeGroup Include="arch">
+      <Parent>linux</Parent>
+      <Architectures>x64</Architectures>
+    </RuntimeGroup>
+
     <RuntimeGroup Include="centos">
       <Parent>rhel</Parent>
       <Architectures>x64</Architectures>
